src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IndexerAccess.cs

```diff
@@ -261,13 +314,22 @@ private BoundSequence VisitIndexOrRangePatternIndexerAccess(BoundIndexOrRangePat
 
             var F = _factory;
 
+            // The argument value may be a non-placeholder value (in an element access scenario: `expr[^1]`or `array[^1]`)
+            // or a placeholder value (in a list-pattern scenario: `expr is [_, var x]` or `array is [_, var x]`).
+            unloweredExpr = UnwrapPlaceholderIfNeeded(unloweredExpr);
```

Comment:
>unloweredExpr = UnwrapPlaceholderIfNeeded(unloweredExpr); [](http://example.com/codeflow?start=12&length=57)

This doesn't look right. This is effectively doing what ```VisitExpression``` would do to a placeholder. That should not be a subject to another ```VisitExpression``` below.